kadai1/shinnosuke-K/file/files.go

```diff
@@ -0,0 +1,73 @@
+package file
+
+import (
+	"fmt"
+	"image"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type File struct {
+	Dir       string // absolute directory path which files exists
+	Name      string // File Name
+	Extension string
+}
+
+var fileList = make([]File, 0)
```

Comment:
これはGetImgFiles内に書くべき（GetImgFilesでしか使わないので）
```suggestion
var fileList []File
```

Reply:
GetImgFiles は再帰的にファイルを探索して、条件に満たすファイルを fileList へ append しています。
関数内で fileList を宣言すると、呼ばれるたびに初期化されます。
再帰的に呼ばれても情報を保持するためにグローバル変数にしています。
関数内に入れるためには処理を変更する必要がありそうです。

Reply:
あれ...何か勘違いしていました...すみません。
クロージャを使えばグローバル変数をなくせそうですが、今のほうが可読性が高い気もします。何かコメントがあると読みやすいかもです